src/install.js

```diff
@@ -54,13 +75,26 @@ async function downloadSAPUI5(downloadURL, downloadPath) {
   console.warn('By using this npm package you agree to the EULA from SAP: https://tools.hana.ondemand.com/developer-license-3_1.txt/')
   console.log('Downloading SAPUI5...')
   const zipFile = path.join(downloadPath, 'sapui5.zip')
+  let response
   try {
-    const response = await axios.get(downloadURL, {
-      responseType: 'stream',
-      headers: {
-        Cookie: 'eula_3_1_agreed=tools.hana.ondemand.com/developer-license-3_1.txt',
-      },
-    })
+    if (agent) {
+      response = await axios.get(downloadURL, {
+        responseType: 'stream',
+        headers: {
+          Cookie: 'eula_3_1_agreed=tools.hana.ondemand.com/developer-license-3_1.txt',
+        },
+        httpsAgent: agent,
+        proxy: false,
+
+      })
+    } else {
+      response = await axios.get(downloadURL, {
+        responseType: 'stream',
+        headers: {
+          Cookie: 'eula_3_1_agreed=tools.hana.ondemand.com/developer-license-3_1.txt',
+        },
+      })
+    }
```

Comment:
As the axios config is quite similar I would prefer something like:
```javascript
const requestConfig = {
  responseType: 'stream',
  headers: {
    Cookie: 'eula_3_1_agreed=tools.hana.ondemand.com/developer-license-3_1.txt',
  },
}

if (agent) {
  requestConfig.httpsAgent = agent
  requestConfig.proxy = false
}

const response = await axios.get(downloadURL, requestConfig)
```

Reply:
Agree, implemented